go/enclave/enclave.go

```diff
@@ -592,19 +589,22 @@ func (e *enclaveImpl) GetTransactionCount(encryptedParams common.EncryptedParams
 		return nil, responses.ToInternalError(fmt.Errorf("requested GetTransactionCount with the enclave stopping"))
 	}
 
-	var nonce uint64
-	paramBytes, err := e.rpcEncryptionManager.DecryptBytes(encryptedParams)
+	// decode the received request into a []interface
+	paramList, err := e.decodeRequest(encryptedParams)
 	if err != nil {
-		return responses.AsPlaintextError(err), nil
+		return responses.AsPlaintextError(fmt.Errorf("unable to decode eth_transactionCount params - %w", err)), nil
 	}
 
-	address, err := rpc.ExtractAddress(paramBytes)
-	if err != nil {
-		return responses.AsPlaintextError(err), nil
+	// Parameters are [Address]
+	if len(paramList) < 1 {
+		return responses.AsPlaintextError(fmt.Errorf("unexpected number of parameters")), nil
 	}
 
+	address := gethcommon.HexToAddress(paramList[0].(string))
```

Comment:
Don't we need to check this cast?

Reply:
It seemed like they weren't checked before. But you're right, they should be checked !